iree/base/wait_handle_poll.c

```diff
@@ -56,13 +56,13 @@ static iree_status_t iree_syscall_poll(struct pollfd* fds, nfds_t nfds,
     // One or more events set.
     *out_signaled_count = rv;
     return iree_ok_status();
-  } else if (rv == 0) {
-    // Timeout; no events set.
-    return iree_status_from_code(IREE_STATUS_DEADLINE_EXCEEDED);
   } else if (IREE_UNLIKELY(rv < 0)) {
     return iree_make_status(iree_status_code_from_errno(errno),
                             "poll failure %d", errno);
   }
+  // rv == 0
+  // Timeout; no events set.
+  return iree_status_from_code(IREE_STATUS_DEADLINE_EXCEEDED);
```

Comment:
Just tested this in the compiler explorer, no diff there (but the `IREE_UNLIKELY` also has no effect in the simplified test case with clang in `-O3`): https://godbolt.org/z/M1xY1G